### PR TITLE
Fix error message when trying to install local NApps

### PR DESCRIPTION
**Pull request by josemauro**
The issue #278 describe a problem installing a local NApp using the kytos tutorial.
After create a NApp using 'kytos napps create'  the attempt to install the created NApp it was returing the following result:

> INFO    NApp test/helloworld:
INFO      Downloading from NApps Server...
ERROR Error installing the NApp.
INFO      Downloaded and installed.
INFO      Enabling...
ERROR NApp is not installed. Check the NApp list.
ERROR     Error enabling NApp.
...
 urllib.error.HTTPError: HTTP Error 400: BAD REQUEST

You can check the issue #278 for the complete output above. To solve this problem it is necessary improve the following:

- Let the module kytos/cli/commands/napps/api.py handle with HTTP code 400. (9a53047)
- Inform the user to install the created NApp using 'setup.py'. (9a53047)  
- Create a template for setup.py to be added when a new NApp is created. (76caa18, a16e539)

### CODE REVIEW

**Comment by josemauro** [comment]
Tip for testing the PR:
- created a new NApp;
- try enable and disable the created NApp;
- uninstall the NApp.

File: kytos/cli/commands/napps/api.py [hunk 1]
```diff
@@ -177,7 +177,10 @@ def install_napp(cls, mgr):
         except HTTPError as exception:
             if exception.code == 404:
                 LOG.error('    NApp not found.')
-            else:
+                LOG.info("    If you are trying to install a local NApp, you h"
+                         "ave to use the command 'python setup.py install' ins"
+                         "ide of the 'user/napp' directory.")
```
Comment (hdiogenes) [inline kytos/cli/commands/napps/api.py:182]:
```suggestion
                LOG.info("    If you are trying to install a local NApp, you "
                         "have to use the command 'python setup.py develop' "
                         "inside the 'user/napp' directory.")
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    raise RuntimeError("This setup.py does not support wheels")
+
+# Paths setup with virtualenv detection
+if 'VIRTUAL_ENV' in os.environ:
+    BASE_ENV = Path(os.environ['VIRTUAL_ENV'])
+else:
+    BASE_ENV = Path('/')
+
+NAPP_NAME = '{{napp}}'
+NAPP_VERSION = '0.0'
```
Comment (hdiogenes) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:28]:
```suggestion
NAPP_VERSION = '0.1'
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+          'develop': DevelopMode,
+          'install': InstallMode,
+          'lint': Linter,
+          'egg_info': EggInfo,
+      },
+      zip_safe=False,
+      classifiers=[
+          'License :: OSI Approved :: MIT License',
+          'Operating System :: POSIX :: Linux',
+          'Programming Language :: Python :: 3.6',
```
Comment (hdiogenes) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:254]:
```suggestion
          'Programming Language :: Python :: 3',
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+
+class CITest(SimpleCommand):
+    """Run all CI tests."""
+
+    description = 'run all CI tests: unit and doc tests, linter'
+
+    def run(self):
+        """Run unit tests with coverage, doc tests and linter."""
+        cmds = ['python3.6 setup.py ' + cmd
```
Comment (hdiogenes) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:102]: Let's start removing the hard-coded 3.6 version references.

```suggestion
        cmds = ['python3 setup.py ' + cmd
```

File: tests/unit/commands/test_napps_api.py [hunk 5]
```diff
@@ -180,6 +182,35 @@ def test_install_napp(self, mock_napps_manager):
 
         mgr.remote_install.assert_called()
 
+    @patch('kytos.utils.napps.NAppsManager')
+    def test_install_napp_error_404(self, mgr):
+        """Test install_napp method."""
+        mgr = MagicMock()
+
+        url = 'www.napps.kytos.io'
```
Comment (hdiogenes) [inline tests/unit/commands/test_napps_api.py:190]: There's no 'www' here.

```suggestion
        url = 'napps.kytos.io'
```

File: tests/unit/commands/test_napps_api.py [hunk 6]
```diff
@@ -180,6 +182,35 @@ def test_install_napp(self, mock_napps_manager):
...
+        mgr.remote_install.side_effect = HTTPError(url, code, msg, hdrs, filep)
+        with self.assertRaises(KytosException):
+            self.napps_api.install_napp(mgr)
+
+    @patch('kytos.utils.napps.NAppsManager')
+    def test_install_napp_error_400(self, mgr):
+        """Test install_napp method."""
+        mgr = MagicMock()
+
+        url = 'www.napps.kytos.io'
```
Comment (hdiogenes) [inline tests/unit/commands/test_napps_api.py:204]:
```suggestion
        url = 'napps.kytos.io'
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+        """Build css."""
+        # self._install_deps_wheels()
+        super().run()
+
+    # @staticmethod
+    # def _install_deps_wheels():
+    #    """Python wheels are much faster (no compiling)."""
+    #    print('Installing dependencies...')
+    #    check_call([sys.executable, '-m', 'pip', 'install', '-r',
+    #                'requirements/run.in'])
```
Comment (hdiogenes) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:135]: I think this class is now deprecated. Also, please take a look at kytos/kytos#1114.

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+from setuptools.command.install import install
+
+if 'bdist_wheel' in sys.argv:
+    raise RuntimeError("This setup.py does not support wheels")
+
+# Paths setup with virtualenv detection
+if 'VIRTUAL_ENV' in os.environ:
+    BASE_ENV = Path(os.environ['VIRTUAL_ENV'])
+else:
+    BASE_ENV = Path('/')
```
Comment (gleybersonandrade) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:25]:
```suggestion
# Paths setup with virtualenv detection
BASE_ENV = Path(os.environ.get('VIRTUAL_ENV', '/'))
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template (hunk 7 above)
Comment (gleybersonandrade) [reply kytos/templates/skel/napp-structure/username/napp/setup.py.template:135]: I think this class can be safely removed

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+        """Build css."""
+        # self._install_deps_wheels()
+        super().run()
+
+    # @staticmethod
+    # def _install_deps_wheels():
+    #    """Python wheels are much faster (no compiling)."""
+    #    print('Installing dependencies...')
+    #    check_call([sys.executable, '-m', 'pip', 'install', '-r',
+    #                'requirements/run.in'])
```
Comment (gleybersonandrade) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:135]:
```suggestion
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+          ],
+      },
+      cmdclass={
+          'clean': Cleaner,
+          'ci': CITest,
+          'coverage': TestCoverage,
+          'develop': DevelopMode,
+          'install': InstallMode,
+          'lint': Linter,
+          'egg_info': EggInfo,
```
Comment (gleybersonandrade) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:248]:
```suggestion
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+import os
+import shutil
+import sys
+from abc import abstractmethod
+from pathlib import Path
+from subprocess import call, check_call
+
+from setuptools import Command, setup
+from setuptools.command.develop import develop
+from setuptools.command.egg_info import egg_info
```
Comment (gleybersonandrade) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:15]:
```suggestion
```

File: kytos/templates/skel/napp-structure/username/napp/setup.py.template [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+        (ENABLED_PATH / '{{username}}').mkdir(parents=True, exist_ok=True)
+        for napp in CORE_NAPPS:
+            napp_path = Path('{{username}}', napp)
+            src = ENABLED_PATH / napp_path
+            dst = INSTALLED_PATH / napp_path
+            symlink_if_different(src, dst)
+
+
+
+
```
Comment (gleybersonandrade) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:119]:
```suggestion
```

Comment (gleybersonandrade) [inline kytos/templates/skel/napp-structure/username/napp/setup.py.template:119]:
```suggestion

```

**Pull request merged**